common/meta/api/src/schema_api_impl.rs

```diff
@@ -227,18 +389,83 @@ impl<KV: KVApi> SchemaApi for KV {
             let (db_id_seq, _db_id) = get_id_value(self, &tenant_newdbname).await?;
             db_has_to_not_exist(db_id_seq, &tenant_newdbname, "rename_database")?;
 
+            // get db id list from _fd_db_id_list/<tenant>/<db_name>
+            let dbid_idlist = DbIdListKey {
+                tenant: tenant_dbname.tenant.clone(),
+                db_name: tenant_dbname.db_name.clone(),
+            };
+            let (db_id_list_seq, db_id_list_opt): (_, Option<DbIdList>) =
+                get_struct_value(self, &dbid_idlist).await?;
+            let mut db_id_list: DbIdList;
+            if db_id_list_seq == 0 {
+                db_id_list = DbIdList::new();
+                db_id_list.append(old_db_id);
```

Comment:
If the db exists but the `db_id_list` does not, it must be a bug and an error should be returned. Unless metasrv allows removing a `db_id_list` record while the db it belongs to still exists.

Reply:
there is a case that: already create a {db|table} before we add time travel related function, thus these {db|table} is not in id list, so we must handle this case.